pdb-slave/MARCH4-PDB/lib/StateMachine/StateMachine.cpp

```diff
@@ -0,0 +1,184 @@
+#include "StateMachine.h"
```

Comment:
This class is a bit long. Below I will give some suggestions to place some code in methods. Those methods can be placed in a different class.